pandas/core/internals/managers.py

```diff
@@ -781,14 +781,16 @@ def copy_func(ax):
         res.axes = new_axes
         return res
 
-    def as_array(self, transpose: bool = False) -> np.ndarray:
+    def as_array(self, transpose: bool = False, na_value=lib.no_default) -> np.ndarray:
```

Comment:
I think we should also pass `dtype` here, to avoid eg fist converting ints/floats to object and then converting back to numeric dtype in `DataFrame.to_numpy`